lib/flake-checks/flake.nix

```
@@ -0,0 +1,35 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+    nix-unit.url = "github:nix-community/nix-unit";
+    nix-unit.inputs.nixpkgs.follows = "nixpkgs";
+  };
+  outputs = inputs: inputs.flake-utils.lib.eachDefaultSystem (system:
+    let
+      pkgs = inputs.nixpkgs.legacyPackages.${system};
+    in
+    {
+      tests.testPass = { expr = 3; expected = 4; };
+
+      checks.default = pkgs.stdenv.mkDerivation {
+        name = "default";
+        phases = [ "unpackPhase" "buildPhase" ];
+        src = inputs.self;
+        nativeBuildInputs = [ inputs.nix-unit.packages.${system}.default ];
+        buildPhase = ''
+          export HOME="$(realpath .)"
+          nix-unit \
+            --eval-store "$HOME" \
+            --flake \
+            --option extra-experimental-features flakes \
```

Maybe a comment for people why we need to override the inputs.
```suggestion
            # The nix derivation must be able to find all used inputs in the nix-store because it cannot download it during buildTime. 
            --override-input nixpkgs ${inputs.nixpkgs.outPath} \
```